sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1870,4 +1870,19 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       assert(sampled.count() == sampledOdd.count() + sampledEven.count())
     }
   }
+
+  test("SPARK-10707: nullability should be correctly propagated through set operations") {
+    withTempTable("src") {
+      Seq((1, 1)).toDF("key", "value").registerTempTable("src")
+      checkAnswer(
+        sql("""SELECT count(v) FROM (
+              |  SELECT v FROM (
+              |    SELECT 'foo' AS v FROM src UNION ALL
+              |    SELECT NULL AS v FROM src
+              |  ) my_union WHERE isnull(v)
+              |) my_subview""".stripMargin),
+        Seq(Row(0)))
```

Comment:
Then what does this test prove? Without this fix, `my_union` is not nullable and `isnull(v)` will always be false, so `my_subview` will be empty and the result of `count(v)` is also 0, did I miss something here?